modin/pandas/test/dataframe/test_default.py

```diff
@@ -1171,3 +1171,15 @@ def test_hasattr_sparse(is_sparse_data):
         else create_test_dfs(test_data["float_nan_data"])
     )
     eval_general(modin_df, pandas_df, lambda df: hasattr(df, "sparse"))
+
+
+def test_setattr_axes():
+    # Test that setting .index or .columns does not warn
+    df = pd.DataFrame([[1, 2], [3, 4]])
+    with pytest.warns(None):
+        df.index = ["foo", "bar"]
```

Comment:
`pytest.warns(None)` [doesn't check](https://stackoverflow.com/q/45671803/17554722) that we're not warning, so this test case doesn't reproduce the bug. The [recommended solution](https://docs.pytest.org/en/7.0.x/how-to/capture-warnings.html#additional-use-cases-of-warnings-in-tests) is to use `warnings.catch_warnings()` and `warnings.simplefilter("error")`. I just checked that that reproduces the bug.

```suggestion
    with warnings.catch_warnings():
        warnings.simplefilter("error")
        df.index = ["foo", "bar"]
```

You'll also have to `import warnings` at the top of the file.

Reply:
could use `pandas.testing.assert_produces_warning(None):`

Reply:
looks like `assert_produces_warning` doesn't live in `pandas.testing` anymore, and it was [moved](https://stackoverflow.com/a/63532540/17554722) to [`pandas/_testing`](https://github.com/pandas-dev/pandas/blob/c9c6685c51ead26bbbb9a0dd565e82967cd839e8/pandas/_testing/_warnings.py#L19). I guess we should use `catch_warnings` after all.